environment/allocations.go

```diff
@@ -38,15 +38,22 @@ func (a *Allocations) Bindings() nat.PortMap {
 				continue
 			}
 
-			binding := []nat.PortBinding{
-				{
-					HostIP:   ip,
-					HostPort: strconv.Itoa(port),
-				},
+			binding := nat.PortBinding{
+				HostIP:   ip,
+				HostPort: strconv.Itoa(port),
 			}
 
-			out[nat.Port(fmt.Sprintf("%d/tcp", port))] = binding
-			out[nat.Port(fmt.Sprintf("%d/udp", port))] = binding
+			if v, ok := out[nat.Port(fmt.Sprintf("%d/tcp", port))]; ok {
```

Comment:
Those `fmt.Sprintf`s should definitely be extracted into variables as you suggested yourself in one of the review comments.

Reply:
The entire `nat.Port()` should be extracted into variables, not just the `fmt.Sprintf`s

Reply:
Yes. It's late here.

Reply:
So together with your previous reply we could make it:
```go
tcpKey := nat.Port(fmt.Sprintf("%d/tcp", port))
udpKey := nat.Port(fmt.Sprintf("%d/udp", port))

out[tcpKey] = append(out[tcpKey], binding)
out[udpKey] = append(out[udpKey], binding)
```

Reply:
Disregard my comment regarding readability though, I'm not sure what best practice here is, but I guess using the returned value is fine.
So keep it as `if v, ok :=` and `out[tcp] = append(v, binding)`.
You could see if `o` instead of `v` is readable, but it might be bad because of the `ok` afterwards :D